kf/nfconfig.go

```diff
@@ -1205,7 +1207,16 @@ func (c *NFConfigs) DeleteProgramsOnInterface(ifaceName, HostName string, bpfPro
 	}
 
 	if _, ok := c.hostInterfaces[ifaceName]; !ok {
-		errOut := fmt.Errorf("%s interface name not found in the host", ifaceName)
+		errOut := fmt.Errorf("%s interface name not found in the host, and deleting all EBPF programs associated to ", ifaceName)
+		if c.EgressTCBpfs[ifaceName] != nil {
+			c.EgressTCBpfs[ifaceName] = nil
```

Comment:
This sets the lists to nil, are there any additional cleanups that need to happen or is this enough to remove the reference counts and allow the garbage collector to lazily clean up?

Reply:
Once the interface is deleted, all the programs associated with will automatically be deleted by the kernel. 
I have added a separate function for cleanup all the L3AFd related parameters and information related to the deleted interface.